SplunkRumWorkspace/SplunkRum/SplunkRum/AppStart.swift

```diff
@@ -42,39 +42,101 @@ private func processStartTime() throws -> Date {
+ We can set this flag to do additional calculation adjustments.
+*/
+var wasBackgroundedBeforeWillEnterForeground: Bool = false
+
 func initializeAppStartupListeners() {
-    let events = [
-        UIApplication.didFinishLaunchingNotification,
-        UIApplication.willEnterForegroundNotification,
-        UIApplication.didBecomeActiveNotification
-    ]
-    var reportedEvents = Set<Notification.Name>()
-    events.forEach { event in
-        _ = NotificationCenter.default.addObserver(forName: event, object: nil, queue: nil) { (_) in
-            if !reportedEvents.contains(event) {
-                reportedEvents.insert(event)
-                if appStart == nil {
-                    return
-                }
-                appStart!.addEvent(name: event.rawValue)
-                if event == UIApplication.didBecomeActiveNotification {
-                    appStart!.end()
-                    OpenTelemetry.instance.contextProvider.removeContextForSpan(appStart!)
-                    appStart = nil
-                    // Because of heavy overlap and desired treatment of AppStart vs
-                    // ongoing app lifecycle stuff, initialize this now rather than
-                    // earlier to avoid double-reporting or more complex logic
-                    initializeAppLifecycleInstrumentation()
-                }
+    let notifCenter = NotificationCenter.default
+
+    // didBecomeActive
+    var didBecomeActiveNotificationToken: NSObjectProtocol?
+    let didBecomeActiveClosure: (Notification) -> Void = { notification in
+        // defering closure scope exit to clean up appStart, start lifecycle instrumentation, and remove observer.
+        defer {
+            appStart = nil
+            // Because of heavy overlap and desired treatment of AppStart vs
+            // ongoing app lifecycle stuff, initialize this now rather than
+            // earlier to avoid double-reporting or more complex logic
+            initializeAppLifecycleInstrumentation()
+
+            if let didBecomeActiveNotificationToken = didBecomeActiveNotificationToken {
+                notifCenter.removeObserver(didBecomeActiveNotificationToken)
             }
         }
+
+        guard let appStart = appStart else { return }
+
+        // If we are prewarmed,
+        // the app was made active from the background,
+        // or we are over a nonsensical threshold, we do not report the appStart span.
+        if (isPrewarm && prewarmAvailable)
+            || wasBackgroundedBeforeWillEnterForeground
+            || (Date().timeIntervalSince1970 - (spanStart.timeIntervalSince1970)) > possibleAppStartTimingErrorThreshold {
+            OpenTelemetry.instance.contextProvider.removeContextForSpan(appStart)
+        } else {
+            appStart.addEvent(name: notification.name.rawValue)
+            appStart.end()
+            OpenTelemetry.instance.contextProvider.removeContextForSpan(appStart)
```

Comment:
I think calling `end` should already remove it: https://github.com/open-telemetry/opentelemetry-swift/blob/main/Sources/OpenTelemetrySdk/Trace/RecordEventsReadableSpan.swift#L281

Reply:
Oh interesting. Will update.